app-modules/prospect/src/Filament/Resources/ProspectResource/Actions/ConvertToStudent.php

```diff
@@ -94,6 +95,8 @@ protected function setUp(): void
                     ->title('Prospect converted to Student')
                     ->success()
                     ->send();
+
+                $this->redirect(ProspectResource::getUrl('view', ['record' => $this->record]));
```

Comment:
You should check whether or not we are actually on the view page or not. Because if you are, then you don't need to redirect.